app/controllers/IndexController.scala

```diff
@@ -101,8 +101,9 @@ class IndexController @Inject()(
 
   override def transformation(implicit request: AuthenticatedRequest[_]): Future[Result] =
     extractViewModel flatMap {
-      case Right(noATS: NoATSViewModel) => Future.successful(Redirect(routes.ErrorController.authorisedNoAts()))
-      case Right(result: ViewModel)     => getViewModel(result)
+      case Right(result: ViewModel)          => getViewModel(result)
+      case Right(_: NoATSViewModel)          => Future.successful(Redirect(routes.ErrorController.authorisedNoAts()))
+      case Right(_: ATSUnavailableViewModel) => Future.successful(InternalServerError(genericErrorView()))
```

Comment:
What happens on the Left case here or any other `GenericViewModel`?

Reply:
code is written in such a way that it always returns Right(_) and is going to have only these three GenericViewModel:
https://github.com/hmrc/tax-summaries-frontend/blob/master/app/controllers/IndexController.scala#L87

Reply:
There are many classes that extend GenericViewModel in this repo so this won't satisfy them all, consider what you would do for other GenericViewModels